src/components/TaskDashboard.tsx

```diff
@@ -1,6 +1,12 @@
 import React from 'react';
+import { ITaskList } from './types';
 
-function TaskDashboard() {
+interface TaskList {
+  tasks: ITaskList[];
+}
+
```

Comment:
Deveria ser:

```suggestion
interface TaskList {
  tasks: Task[];
}
```

O tipo ItaskList[] que você criou é uma lista de objetose e nesse objeto existe uma chave uuid com um objeto task

No final o objeto que você estava trabalhando ficava neste formato:
```javascript
[
  {
    12931923123: {
      id: '12931923123',
      textTask: 'test',
      taskComplete: false,
    },
  },
  {
    123123: {
      id: '123123',
      textTask: 'test',
      taskComplete: false,
    },
  },
];
```
Perceba que o ID é redundante e temos uma nova camada de objeto completamente desnecessária
O dado final, após minhas indicações seria:

```javascript
[
  {
    id: '12931923123',
    textTask: 'test',
    taskComplete: false,
  },
  {
    id: '12931231231923123',
    textTask: 'test',
    taskComplete: false,
  },
];
```

Veja como faz mais sentido e simplifica o acesso as tasks (veja meus outros comments sobre o tema)